core/ui/mpc/keysign/KeysignSigningStep.tsx

```
@@ -37,123 +46,218 @@ export const KeysignSigningStep = ({ onBack }: KeysignSigningStepProps) => {
+    if (keysignPayload.swapPayload && keysignPayload.swapPayload.value) return
+    // Check if any transaction has a successful scan result
+    const hasSuccessfulScan = txResults.some(
+      txResult =>
+        txResult.scanResult?.validation?.status === 'Success' &&
+        txResult.scanResult?.validation?.result_type ===
+          BlockaidResultTypes.Benign
+    )
+    setShowSecurityCheckmark(hasSuccessfulScan)
+    // Check if scanUnavailable is true in the result
+    setScanUnavailable(
+      !!(txResults && txResults[0] && txResults[0].scanUnavailable)
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)